### DISCUSSION TITLE
Solution for (Failed to install solc 0.8.20)

### DISCUSSION THREAD

**Discussion by jyotirmoydotdev** [Cyfrin/foundry-full-course-cu#85] [Ideas]
Just want to let you know that if you are facing an error while running `forge create` command for your smart contract project, there is a solution for you! The error message might look something like this:

```bash
➜  Smart-Contract-Foundry git:(main) ✗ forge create Counter --interactive
[⠊] Compiling...
[⠃] Installing solc version 0.8.20
[⠊] Failed to install solc 0.8.20: error sending request for url (<https://raw.githubusercontent.com/alloy-rs/solc-builds/5560a1f4b69f081d120b0df1735bf87cdda2e6b5/macosx/aarch64/list.json>): error trying to connect: tcp connect error: Operation timed out (os error 60)
Error:
error sending request for url (<https://raw.githubusercontent.com/alloy-rs/solc-builds/5560a1f4b69f081d120b0df1735bf87cdda2e6b5/macosx/aarch64/list.json>): error trying to connect: tcp connect error: Operation timed out (os error 60)

Context:
- Error #0: error trying to connect: tcp connect error: Operation timed out (os error 60)
- Error #1: tcp connect error: Operation timed out (os error 60)
- Error #2: Operation timed out (os error 60)
```

# solution

First, install solidity with the following command:

```bash
brew update
brew upgrade
brew tap ethereum/ethereum
brew install solidity
```

Once you have installed solidity, copy the installed directory address which you get with the command `brew info solidity`. You only need the path up to the bin directory. For example:

```bash
/opt/homebrew/Cellar/solidity/0.8.20/bin
```

Next, navigate to the main directory of your project and open the `foundry.toml` file. This file contains configuration details for your project. Add the following line to the file:

```toml
solc = "<path-to-solc>"
```

Replace `<path-to-solc>` with the bin directory path you copied earlier. For example:

```toml
solc = "/opt/homebrew/Cellar/solidity/0.8.20/bin/solc"
```

The full code should look like this.

```toml
[profile.default]
src = "src"
out = "out"
libs = ["lib"]
solc = "/opt/homebrew/Cellar/solidity/0.8.20/bin/solc"

# See more config options https://github.com/foundry-rs/foundry/tree/master/config
```

Save the file and exit. Now try running the `forge create` command again. It should run without any errors.

By following these steps, you should be able to successfully resolve the `solc` installation error and create your smart contract project with ease. Good luck coding!

**Comment by PatrickAlphaC** [comment]
Thank you for this!
